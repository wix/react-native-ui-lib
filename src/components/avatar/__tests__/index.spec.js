import React from 'react';
import {StyleSheet} from 'react-native';
import {render} from '@testing-library/react-native';
import {Avatar} from '../index';

function verifyBadgeSize(renderTree, expectedSize) {
  const badge = renderTree.getByTestId('avatar.onlineBadge');
  const style = StyleSheet.flatten(badge.props.style);
  expect(style.width).toEqual(expectedSize);
  expect(style.height).toEqual(expectedSize);
}

describe('Avatar Badge', () => {
  describe('badgeProps.size supports number', () => {
    it('should return 99 as the size number given', () => {
      const renderTree = render(<Avatar testID={'avatar'} badgeProps={{size: 99}}/>);
      verifyBadgeSize(renderTree, 99);
    });

    it('should return default when passing 0 as size', () => {
      const renderTree = render(<Avatar testID={'avatar'} badgeProps={{size: 0}}/>);
      verifyBadgeSize(renderTree, 10);
    });
  });
});

describe('Avatar Accessibility', () => {
  it('should render with hitSlop when onPress is provided', () => {
    const renderTree = render(<Avatar testID={'avatar'} onPress={() => {}}/>);
    expect(renderTree.toJSON()).toMatchSnapshot();
  });

  it('should not render hitSlop when onPress is not provided', () => {
    const renderTree = render(<Avatar testID={'avatar'}/>);
    expect(renderTree.toJSON()).toMatchSnapshot();
  });
});
