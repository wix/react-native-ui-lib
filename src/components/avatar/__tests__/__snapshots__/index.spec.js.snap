// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Accessibility should not render hitSlop when onPress is not provided 1`] = `
<View
  accessibilityLabel="Avatar"
  accessibilityRole="image"
  accessible={false}
  style={
    [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      [
        {
          "alignItems": "center",
          "borderRadius": 999,
          "height": 50,
          "justifyContent": "center",
          "width": 50,
        },
        undefined,
      ],
    ]
  }
  testID="avatar"
>
  <View
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "alignItems": "center",
            "borderRadius": 999,
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#F8f9FA",
          },
          false,
        ],
      ]
    }
    testID="avatar.container"
  />
</View>
`;

exports[`Avatar Accessibility should render with hitSlop when onPress is provided 1`] = `
<View
  accessibilityLabel="Avatar"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    {
      "bottom": 0,
      "left": 0,
      "right": 0,
      "top": 0,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 999,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 50,
    }
  }
  testID="avatar"
>
  <View
    style={
      [
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        [
          {
            "alignItems": "center",
            "borderRadius": 999,
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "backgroundColor": "#F8f9FA",
          },
          false,
        ],
      ]
    }
    testID="avatar.container"
  />
</View>
`;
